rholang/src/main/scala/coop/rchain/rholang/interpreter/storage/ChargingRSpace.scala

```diff
@@ -39,14 +41,24 @@ object ChargingRSpace {
           patterns: Seq[BindPattern],
           continuation: TaggedContinuation,
           persist: Boolean,
-          sequenceNumber: Int
+          sequenceNumber: Int,
+          peek: Boolean
       ): F[
         Option[(ContResult[Par, BindPattern, TaggedContinuation], Seq[Result[ListParWithRandom]])]
       ] =
         for {
-          _       <- charge[F](storageCostConsume(channels, patterns, continuation))
-          consRes <- space.consume(channels, patterns, continuation, persist, sequenceNumber)
-          _       <- handleResult(consRes)
+          _ <- charge[F](storageCostConsume(channels, patterns, continuation))
+          consRes <- space.consume(
+                      channels,
+                      patterns,
+                      continuation,
+                      persist,
+                      sequenceNumber,
+                      if (peek) {
+                        SortedSet((0 to channels.size - 1): _*)
+                      } else SortedSet.empty
```

Comment:
is there a ticket for expanding the `for` syntax in rholang to accommodate mixed consumes?

Reply:
I don't think there is one, we should discuss whether this is needed for mercury (or at all)

Reply:
I didn't assume that, so maybe we should have been more explicit about this. From my perspective the amount of changes needed to implement thist would fan-out wide, and as it isn't absolutely necessary it could be done in a separate PR. Here's the ticket:
https://rchain.atlassian.net/browse/RCHAIN-3607